crates/wasi/src/preview2/tcp.rs

```diff
@@ -214,6 +217,20 @@ impl HostOutputStream for TcpWriteStream {
 }
 
 impl HostTcpSocket {
+    // The following DEFAULT_BACKLOG logic is from
+    // <https://github.com/rust-lang/rust/blob/master/library/std/src/sys_common/net.rs>
+    // at revision defa2456246a8272ceace9c1cdccdf2e4c36175e.
+
+    // The 3DS doesn't support a big connection backlog. Sometimes
+    // it allows up to about 37, but other times it doesn't even
+    // accept 32. There may be a global limitation causing this.
+    #[cfg(target_os = "horizon")]
+    const DEFAULT_BACKLOG: i32 = 20;
```

Comment:
To avoid repeating the definition could this be:

```rust
const DEFAULT_BACKLOG: i32 = if cfg!(target_os = "horizon") { 20 } else { 128 };
```

Reply:
Or, alternatively, could the default be `None`? That way we could let the OS defaults come into play if it's not otherwise specified.

Reply:
That's indeed simpler. I've update the code.